File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        size = len(heights)
        for index in range(size):
            # flag = True
            for val in range(size -index-1):
                if heights[val] < heights[val+1]:
                    flag = False
                    names[val], names[val+1] = names[val+1], names[val]
                    heights[val+1], heights[val] = heights[val], heights[val+1]
            # if flag:
            #     break
                    
        return names

File: 2418-sort-the-people/2418-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        int size = names.size();
        for (int index = 0; index < size; index++){
            bool flag = true;
            for (int val = 0; val < size - index-1; val++){
                if (heights[val] < heights[val+1]){
                    int temp = heights[val];
                    string name = names[val];
                    names[val] = names[val+1];
                    names[val+1] = name;
                    heights[val] = heights[val+1];
                    heights[val+1] = temp;
                    flag = false;
                }
            }
            if (flag){
                break;
            }
        }
        return names;
    }
};